### ISSUE TITLE
Top-scope variable warning on nested each loops

### ISSUE THREAD

**Issue by steventwheeler** [rodjek/puppet-lint#548]
When running puppet-lint on a class which contains nested each loops a top-scope warning is issued. Below is a sample puppet class which can be used to reproduce this problem.
### test.pp

``` puppet
class test {
  [].each |$v1| {
    [].each |$v2| {
      notice("Values: ${v1}, ${v2}")
    }
  }
}
```

```
$ puppet-lint test.pp --no-autoloader_layout-check  --no-documentation-check --with-context
WARNING: top-scope variable being used without an explicit namespace on line 4

  notice("Values: ${v1}, ${v2}")
                   ^
```

The warning is regarding the reference to `$v1` on line 4, the reference to `$v2` does not generate a warning.

```
$ puppet-lint --version
puppet-lint 2.0.2
```

**Comment by rnelson0** [comment]
Fixed by #637
